### PR TITLE
Feature/cloudformation exporting

### PR DESCRIPTION
**Pull request by matarpeles**
1. **Cloudformation Support** -  Added a new handler for CloudFormation Stacks information using the 'cloudformation' boto3 api.
2. **New Structure** - Restructured the Classes, ResourcesHandler generates a new handler dynamically based on 
3. **BaseHandler** - Created a base abstract class for easy implementation of a new AWS handler implentation.

### CODE REVIEW

File: lambda_function/aws/resources/handler_creator.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+from typing import Dict, Type
+
+
+SPECIAL_AWS_HANDLERS:   Dict[str, Type[BaseHandler]] = {
+ "AWS::CloudFormation::Stack": CloudFormationHandler
+}
+
+
+def create_resource_handler(resource_config, port_client, lambda_context, default_region):
+    if resource_config['kind'] in SPECIAL_AWS_HANDLERS:
```
Comment (talsabagport) [inline lambda_function/aws/resources/handler_creator.py:13]: A suggestion to improve:
```
handler = SPECIAL_AWS_HANDLERS.get(resource_config['kind'], CloudControlHandler)
return handler(resource_config, port_client, lambda_context, default_region)
```

File: lambda_function/aws/resources/cloudformation_handler.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import copy
```
Comment (talsabagport) [inline lambda_function/aws/resources/cloudformation_handler.py:1]: Run some formatter / linter

File: lambda_function/aws/resources/cloudformation_handler.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                stack_obj['StackResources'] = aws_cloudformation_client.describe_stack_resources(StackName=stack_id)\
+                    .get('StackResources')
+                stack_obj['Url'] = self._get_stack_console_url(stack_obj['StackId'])
+                template = aws_cloudformation_client.get_template(StackName=stack_id,
+                                                                  TemplateStage='Original')\
+                                                                  .get('TemplateBody')
+
+                # If template is a SAM proccessed JSON, converts to yaml
+                if isinstance(template,dict) or isinstance(template,OrderedDict):
+                    template = yaml.dump(json.loads(json.dumps(template)))
```
Comment (talsabagport) [inline lambda_function/aws/resources/cloudformation_handler.py:78]: Add some comment to explain all the loads / dumps purpose

File: lambda_function/aws/resources/cloudformation_handler.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        try:
+            stack_obj = {}
+            if action_type == 'upsert':
+                logger.info(f"Get CloudFormation Stack, id: {stack_id}")
+
+                aws_cloudformation_client = boto3.client("cloudformation", region_name=region)
+                stack_obj = aws_cloudformation_client.describe_stacks(StackName=stack_id).get("Stacks")[0]
+                stack_obj['StackResources'] = aws_cloudformation_client.describe_stack_resources(StackName=stack_id)\
+                    .get('StackResources')
+                stack_obj['Url'] = self._get_stack_console_url(stack_obj['StackId'])
```
Comment (talsabagport) [inline lambda_function/aws/resources/cloudformation_handler.py:71]: Let's get rid of the URL, and do it in the config

**Review by talsabagport — CHANGES_REQUESTED** [review]

File: lambda_function/aws/resources/cloudformation_handler.py (hunk 3 above)
Comment (matarpeles) [reply lambda_function/aws/resources/cloudformation_handler.py:77]: Try to make it more clear, let me know if it's better now

**Review by talsabagport — APPROVED** [review]

**Pull request merged**